src/dftbp/dftbplus/parser.F90

```diff
@@ -2744,6 +2745,24 @@ subroutine getEuclideanKSampling(tBadIntegratingKPoints, ctrl, node, geo)
       end if
       ctrl%nKPoint = size(ctrl%kPoint, dim=2)
 
+    case ("monkhorstpack")
+      call getChildValue(value1, "Repeat", coeffs)
+      call getChildValue(value1, "GammaCentered", gammaCentered, .false.)
+      if (gammaCentered) then
+        shifts = 0.0_dp
+      else
+        shifts = merge(0.5_dp, 0.0_dp, mod(coeffs, 2) == 0)
```

Comment:
Not sure if this logic is generally correct, as while most unit cells should have a shift of .5,.5,.5 but hexagonal cases are need to be shifted by 0,0,.5 to retain the symmetry for even grids

Reply:
Means we have to detect the space group of the lattice first. Guess that is simple enough to do for the hexagonal case.